File: SimpleCalculator/app/src/main/java/com/aydanilozyurek/simplecalculator/MainActivity.java
package com.aydanilozyurek.simplecalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    // sürekli kullanılacağı için global olarak tanımladık çünkü neden sürekli copy paste yapalım
    EditText number1text;
    EditText number2text;
    TextView resultText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        number1text = findViewById(R.id.number1Text);
        number2text = findViewById(R.id.number2Text);
        resultText = findViewById(R.id.resultText);

    }

    public void sum(View view){

        if(number1text.getText().toString().matches("") || number2text.getText().toString().matches("")){
            resultText.setText("Enter Number!");
        }else{
            int num1 = Integer.parseInt(number1text.getText().toString());
            int num2 = Integer.parseInt(number2text.getText().toString());
            int result = num1 + num2;
            resultText.setText("Result: " + result);
        }


    }
    public void divide(View view){
        int num1 = Integer.parseInt(number1text.getText().toString());
        int num2 = Integer.parseInt(number2text.getText().toString());
        int result = num1 / num2;
        resultText.setText("Result: " + result);
    }
    public void multiply(View view){
        int num1 = Integer.parseInt(number1text.getText().toString());
        int num2 = Integer.parseInt(number2text.getText().toString());
        int result = num1 * num2;
        resultText.setText("Result: " + result);
    }
    public void substraction(View view){
        int num1 = Integer.parseInt(number1text.getText().toString());
        int num2 = Integer.parseInt(number2text.getText().toString());
        int result = num1 - num2;
        resultText.setText("Result: " + result);
    }


}

File: KotlinSimpleCalculator/app/src/main/java/com/aydanilozyurek/kotlinsimplecalculator/MainActivity.kt
package com.aydanilozyurek.kotlinsimplecalculator

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.aydanilozyurek.kotlinsimplecalculator.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    var number1 :Double? = null
    var number2 :Double? = null
    var result :Double? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)


    }

    fun sum(view : View){
         number1 = binding.firstNum.text.toString().toDoubleOrNull()
         number2 = binding.secondNum.text.toString().toDoubleOrNull()

        if(number1 != null && number2 !=null){
             result = number1!! + number2!!
            binding.resultText.text = "Result: ${result}"
        } else {
            binding.resultText.text = "Enter invalid number!"
        }
    }
    fun divide(view : View){

        number1 = binding.firstNum.text.toString().toDoubleOrNull()
        number2 = binding.secondNum.text.toString().toDoubleOrNull()

        if(number1 != null && number2 !=null){
            result = number1!! / number2!!
            binding.resultText.text = "Result: ${result}"
        } else {
            binding.resultText.text = "Enter valid number!"
        }
    }
    fun multiply(view : View){

        number1 = binding.firstNum.text.toString().toDoubleOrNull()
        number2 = binding.secondNum.text.toString().toDoubleOrNull()

        if(number1 != null && number2 !=null){
            result = number1!! * number2!!
            binding.resultText.text = "Result: ${result}"
        } else {
            binding.resultText.text = "Enter invalid number!"
        }
    }
    fun extract(view : View){

        number1 = binding.firstNum.text.toString().toDoubleOrNull()
        number2 = binding.secondNum.text.toString().toDoubleOrNull()

        if(number1 != null && number2 !=null){
            result = number1!! - number2!!
            binding.resultText.text = "Result: ${result}"
        } else {
            binding.resultText.text = "Enter invalid number!"
        }
    }
}
